src/controllers/legacycontrollermapping.h

```diff
@@ -143,6 +210,55 @@ class LegacyControllerMapping {
         return m_deviceDirection;
     }
 
+#ifdef MIXXX_USE_QML
+    /// Adds a custom QML module file to the list of controller modules for this mapping.
+    /// @param dirInfo A FileInfo of the directory or QML module
+    /// @param builtin If this is true, the script won't be written to the XML
+    virtual void addModule(const QFileInfo& dirInfo,
+            bool builtin = false) {
+        m_modules.append(QMLModuleInfo(
+                dirInfo,
+                builtin));
+        setDirty(true);
+    }
+
+    const QList<QMLModuleInfo>& getModules() const {
+        return m_modules;
+    }
+
+    /// @brief Adds a screen info where QML will be rendered.
+    /// @param identifier The screen identifier
+    /// @param size the size of the screen
+    /// @param targetFps the maximum FPS to render
+    /// @param splashoff the rendering grace time given when the screen is requested to shutdown
+    /// @param pixelFormat the pixel encoding format
+    /// @param endian the pixel endian format
+    /// @param reversedColor whether or not the RGB is swapped BGR
+    /// @param rawData whether or not the screen is allowed to reserve bare data, not transformed
+    virtual void addScreenInfo(const QString& identifier,
+            const QSize& size,
+            uint targetFps,
+            uint splashoff,
+            QImage::Format pixelFormat,
+            std::endian endian,
+            bool reversedColor,
+            bool rawData) {
+        m_screens.append(ScreenInfo(identifier,
```

Comment:
It was a hard one for me to make my mind on, but since I've put in `ScreenInfo` in `LegacyControllerMapping`, legacy way of defining a mapping file, I thought it be best not to have more reference to this (hopefully) due to be refactored mapping definition. Arguable, `ScreenInfo` being new code, it will likely remain as is and just be renamed/moved somewhere else, but for now I thought it's best to keep controller mapping component together, and avoid having thing all round the place AND not referenced API marked `Legacy`

Reply:
I think its fair to move the definition of screeninfo to some other non-legacy class.

Reply:
Where would you see that? That's what I meant by having a hard time: because everything is marked "legacy" but not much of the new controller bed has been defined, there is no clear leading space, and instead of putting it alone in a file, I thought best to keep everything parse from XML together, as it is currently the case in this file.

Reply:
Idk. ControllerScriptEngineBase its fine. could also be its own file. Lets not overthink it. there are far greater issues with this PR.

Reply:
Cool, I will keep it here then.